modules/core/src/main/java/org/apache/ignite/internal/util/PointerWrapping.java

```diff
@@ -17,19 +17,268 @@
 
 package org.apache.ignite.internal.util;
 
+import static org.apache.ignite.internal.util.GridUnsafe.NATIVE_BYTE_ORDER;
+import static org.apache.ignite.internal.util.GridUnsafe.UNSAFE;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
 import java.nio.ByteBuffer;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
 
 /**
  * Wraps a pointer to unmanaged memory into a direct byte buffer.
  */
-@SuppressWarnings("InterfaceMayBeAnnotatedFunctional")
-interface PointerWrapping {
+abstract class PointerWrapping {
+    /** Null object. */
+    private static final Object NULL_OBJ = null;
+
+    private static final MethodHandle DIRECT_BUF_MTD;
```

Comment:
Please add some javadocs about what these method handles correspond to

Reply:
I just noticed that `JavaNioAccess` can be used directly in the code, because `jdk.internal.misc` is exported to compiler. That's funny, but I'll leave the old code just in case.

Or should I refactor it?